feature_engine/creation/distance_features.py

```diff
@@ -0,0 +1,241 @@
+    two points on the Earth. However, it is precise enough for our purposes and is fast.
+
+    DistanceFeatures() requires a list of column names of coordinates, i.e., a list of
+    lists of 4 elements, where each 4-list represents the column names of the pair of
+    coordinates for which we should compute the distance. Additionally, it is possible
+    to provide the names of the columns contaning the distances and chose if the
+    coordinate columns are dropped or not.
+
+    Missing data should be imputed before using this transformer.
+
+
+
+    Parameters
+    ----------
+    coordinate_columns: List[List[Union[str, int]]],
+
+    output_column_names: List[Union[str, None]], default=None
+        List of names for the column with the computed distance. Note that the list
+        must have equal length to the `coordinate_columns` list. This is because the
+        transformer need to know which distance column has which name. If none, the
+        default names are generated.
+
+    drop_original: Optional[bool], default=False
+        If True, then the `coordinate_columns` columns are dropped. Otherwise,
+        they are left in the dataframe.
+
+    Attributes
+    ----------
+    ...
+
+    Methods
+    -------
+    fit:
+        Learns the variable's maximum values.
+
+    transform:
+        Compute the distance using the coordinates provided in the `coordinate_columns`.
+
+    References
+    ----------
+    https://en.wikipedia.org/wiki/Haversine_formula
+
+    Examples
+    --------
+
+    >>> import pandas as pd
+    >>> from feature_engine.creation import DistanceFeatures
+    >>> X = pd.DataFrame({
+            'a_latitude': [0., 0., 46.948579],
+            'a_longitude': [0., 0., 7.436925],
+            'b_latitude': [0., 12.34, 59.91054],
+            'b_longitude': [0., 123.45, 10.752695],
+        })
+    >>> cf = DistanceFeatures(
+        coordinate_columns=[['a_latitude', 'a_longitude', 'b_latitude', 'b_longitude']],
+        output_column_names=['distance_between_a_and_b'],
+        drop_original=False,
+    )
+    >>> cf.fit(X)
+    >>> cf.transform(X)
+       a_latitude   a_longitude   b_latitude   b_longitude     distance_between_a_and_b
+    0       0.           0.           0.           0.                   0.
+    1       0.           0.          12.34       123.45             13630.28
+    2      46.94         7.43        59.91        10.75              1457.49
+    """
+
+    _EARTH_RADIUS: float = 6371.0  # radius of Earth in kms
+
+    def __init__(
+        self,
+        coordinate_columns: List[List[Union[str, int]]],
+        output_column_names: Optional[List[Union[str, None]]] = None,
+        drop_original: bool = False,
+    ) -> None:
+
+        # the coordinate_columns variable is rewritten in this way to speed up
+        # computation later, i.e., to use vectorization
+        (self.a_latitudes, self.a_longitudes, self.b_latitudes, self.b_longitudes) = (
```

Comment:
In the init, we check that the user enters allowed values, and then we assign those values to the parameters. We can't have any more functionality than that. 

The unpacking needs to be in the fit() part of the class.